File: package/src/themes.js
// Color themes for text-oriented controls, for use when coupled with a parsing grammar.

// A dark background with a light foreground for text.
export const Dark = Object.freeze({
    name: "Dark",
    cursorColor: "white",
    lineNumbers: {
        foreColor: "white"
    },
    regular: {
        backColor: "black",
        foreColor: "#c0c0c0",
        currentRowBackColor: "#202020",
        selectedBackColor: "#404040",
        unfocused: "rgba(0, 0, 255, 0.25)"
    },
    strings: {
        foreColor: "#aa9900",
        fontStyle: "italic"
    },
    regexes: {
        foreColor: "#aa0099",
        fontStyle: "italic"
    },
    numbers: {
        foreColor: "green"
    },
    comments: {
        foreColor: "yellow",
        fontStyle: "italic"
    },
    keywords: {
        foreColor: "cyan"
    },
    functions: {
        foreColor: "brown",
        fontWeight: "bold"
    },
    members: {
        foreColor: "green"
    },
    error: {
        foreColor: "red",
        fontStyle: "underline italic"
    }
});


// A light background with dark foreground text.
export const Light = Object.freeze({
    name: "Light",
    cursorColor: "black",
    lineNumbers: {
        foreColor: "black"
    },
    regular: {
        backColor: "white",
        foreColor: "black",
        currentRowBackColor: "#f0f0f0",
        selectedBackColor: "#c0c0c0",
        unfocused: "rgba(0, 0, 255, 0.25)"
    },
    strings: {
        foreColor: "#aa9900",
        fontStyle: "italic"
    },
    regexes: {
        foreColor: "#aa0099",
        fontStyle: "italic"
    },
    numbers: {
        foreColor: "green"
    },
    comments: {
        foreColor: "grey",
        fontStyle: "italic"
    },
    keywords: {
        foreColor: "blue"
    },
    functions: {
        foreColor: "brown",
        fontWeight: "bold"
    },
    members: {
        foreColor: "green"
    },
    error: {
        foreColor: "red",
        fontStyle: "underline italic"
    }
});

export const themes = Object.freeze(new Map([
    ["light", Light],
    ["dark", Dark]
]));

File: csharp/Primrose.NETStandard2/Themes.cs
using System.Collections.Generic;
using System.Drawing;

namespace Primrose
{
    public static class Themes
    {
        /// <summary>
        /// A dark background with light foreground text.
        /// </summary>
        public static readonly Theme Dark = new Theme
        {
            name = "Dark",
            cursorColor = Color.White,
            unfocused = Color.FromArgb(0x3f, 0, 0, 0xff),
            currentRowBackColor = Color.FromArgb(0x20, 0x20, 0x20),
            selectedBackColor = Color.FromArgb(0x40, 0x40, 0x40),
            lineNumbers = new ThemeRule
            {
                foreColor = Color.White
            },
            regular = new ThemeRule
            {
                backColor = Color.Black,
                foreColor = Color.FromArgb(0xc0, 0xc0, 0xc0)
            },
            strings = new ThemeRule
            {
                foreColor = Color.FromArgb(0xaa, 0x99, 0x00),
                fontStyle = FontStyle.Italic
            },
            regexes = new ThemeRule
            {
                foreColor = Color.FromArgb(0xaa, 0x00, 0x99),
                fontStyle = FontStyle.Italic
            },
            numbers = new ThemeRule
            {
                foreColor = Color.Green
            },
            comments = new ThemeRule
            {
                foreColor = Color.Yellow,
                fontStyle = FontStyle.Italic
            },
            keywords = new ThemeRule
            {
                foreColor = Color.Cyan
            },
            functions = new ThemeRule
            {
                foreColor = Color.Brown,
                fontStyle = FontStyle.Bold
            },
            members = new ThemeRule
            {
                foreColor = Color.Green
            },
            error = new ThemeRule
            {
                foreColor = Color.Red,
                fontStyle = FontStyle.Italic | FontStyle.Underline
            }
        };


        /// <summary>
        /// A light background with dark foreground text.
        /// </summary>
        public static readonly Theme Light = new Theme
        {
            name = "Light",
            cursorColor = Color.Black,
            unfocused = Color.FromArgb(0x3f, 0, 0, 0xff),
            currentRowBackColor = Color.FromArgb(0xf0, 0xf0, 0xf0),
            selectedBackColor = Color.FromArgb(0xc0, 0xc0, 0xc0),
            lineNumbers = new ThemeRule
            {
                foreColor = Color.Black
            },
            regular = new ThemeRule
            {
                backColor = Color.White,
                foreColor = Color.Black
            },
            strings = new ThemeRule
            {
                foreColor = Color.FromArgb(0xaa, 0x99, 0x00),
                fontStyle = FontStyle.Italic
            },
            regexes = new ThemeRule
            {
                foreColor = Color.FromArgb(0xaa, 0x00, 0x99),
                fontStyle = FontStyle.Italic
            },
            numbers = new ThemeRule
            {
                foreColor = Color.Green
            },
            comments = new ThemeRule
            {
                foreColor = Color.Gray,
                fontStyle = FontStyle.Italic
            },
            keywords = new ThemeRule
            {
                foreColor = Color.Blue
            },
            functions = new ThemeRule
            {
                foreColor = Color.Brown,
                fontStyle = FontStyle.Bold
            },
            members = new ThemeRule
            {
                foreColor = Color.Green
            },
            error = new ThemeRule
            {
                foreColor = Color.Red,
                fontStyle = FontStyle.Italic | FontStyle.Underline
            }
        };

        public static readonly Dictionary<string, Theme> themes = new Dictionary<string, Theme>() {
            { "light", Light },
            { "dark", Dark }
        };
    }
}